File: scratch/replace_colors.js
const fs = require('fs');
const path = require('path');

const baseDir = "c:/Users/MOXIE/Desktop/Vesti";

const replacements = [
    { old: /#060E42/gi, new: '#13110f' },
    { old: /#C0C71E/gi, new: '#00c758' },
    { old: /#62943B/gi, new: '#00a544' },
    { old: /#63A146/gi, new: '#00a544' },
    { old: /#F4FAF2/gi, new: '#f9f5f0' }
];

function processFile(fullPath) {
    let content = fs.readFileSync(fullPath, 'utf8');
    let originalContent = content;
    
    replacements.forEach(rep => {
        content = content.replace(rep.old, rep.new);
    });
    
    if (content !== originalContent) {
        fs.writeFileSync(fullPath, content, 'utf8');
        console.log("Updated colors in: " + fullPath);
    }
}

function traverse(dir) {
    if (!fs.existsSync(dir)) return;
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (file !== 'node_modules' && file !== '.git' && file !== 'scratch') {
                traverse(fullPath);
            }
        } else if (file.endsWith('.html') || file.endswith('.css') || file.endswith('.js')) {
            processFile(fullPath);
        }
    }
}

traverse(baseDir);
console.log("Color replacement complete.");

File: scratch/replace_colors.py
import os
import re

base_dir = "c:/Users/MOXIE/Desktop/Vesti"

replacements = {
    # Old Dark Blue -> Charcoal
    r'#060E42': '#13110f',
    
    # Old Light Green -> WeVesti Light Green
    r'#C0C71E': '#00c758',
    
    # Old Dark Green / Accent Greens -> WeVesti Dark Green
    r'#62943B': '#00a544',
    r'#63A146': '#00a544',
    
    # Old Background Green-White -> WeVesti Cream (or White)
    r'#F4FAF2': '#f9f5f0',
}

def replace_in_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        original_content = content
        
        for old, new in replacements.items():
            # Case insensitive replacement for the hex codes
            content = re.sub(old, new, content, flags=re.IGNORECASE)
            
        if content != original_content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Updated colors in: {filepath}")
            
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

def traverse(directory):
    for root, dirs, files in os.walk(directory):
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        if '.git' in dirs:
            dirs.remove('.git')
        if 'scratch' in dirs:
            dirs.remove('scratch')
            
        for file in files:
            if file.endswith('.html') or file.endswith('.css') or file.endswith('.js'):
                filepath = os.path.join(root, file)
                replace_in_file(filepath)

if __name__ == '__main__':
    traverse(base_dir)
    print("Color replacement complete.")
